### PR TITLE
Providing better workflow for repo + autotagging on travis-ci

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by DaSchTour** [comment]
I would suggest to call this 1.4 and not 1.3.1
This ist a major change in the repos structure

**Comment by ghost** [comment]
ok, changed to `1.4.0`

File: src/smart-app-banner.scss [hunk 1]
```diff
@@ -0,0 +1,319 @@
+//colors
```
Comment (dy) [inline src/smart-app-banner.scss:1]: How `scss` relate to better workflow?
Not sure we can accept that change — no one is going to maintain scss files.

File: src/smart-app-banner.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
+var extend = require('xtend/mutable');
```
Comment (dy) [inline src/smart-app-banner.js:1]: What was the reason to place files into `./src` folder, except for a habit to do so?

File: package.json [hunk 3]
```diff
@@ -1,24 +1,12 @@
...
-  "version": "1.3.0",
-  "description": "Smart banner for Android or iOS",
-  "main": "index.js",
-  "scripts": {
-    "build": "browserify index.js -s SmartBanner | ccjs - > smart-app-banner.js"
-  },
-  "repository": {
-    "type": "git",
-    "url": "git://github.com/kudago/smart-app-banner.git"
+  "version": "%version%",
```
Comment (dy) [inline package.json:3]: Not convinced that is a good practice. What if I do `npm version patch`?

**Comment by dy** [comment]
Hi @piotr-zuralski! Thank you for your contribution.
I am quite confused with the PR though tbh.
Although some changes are valuable, like `travis` and `readme`, I am not sure whether we can find someone to maintain new workflow.

File: src/smart-app-banner.scss (hunk 1 above)
Comment (ghost) [reply src/smart-app-banner.scss:1]: for `scss` files there's an easier way of linting files and generating better `css` files from them, but we can skipp that and remove it

File: src/smart-app-banner.js (hunk 2 above)
Comment (ghost) [reply src/smart-app-banner.js:1]: better readability of project and also good practise

File: package.json (hunk 3 above)
Comment (ghost) [reply package.json:3]: that's just for the `master` branch, if new version would be released only content in `version.txt` must be changed, after accepting PR travis will build new version, if succeeded, `%version%` in files is replaced with that one in `version.txt` and created new tag with that version.

**Comment by ghost** [comment]
Hi @dfcreative,
I can describe more then proposed workflow to be more clear about steps and purposes, if you want.

File: src/smart-app-banner.js (hunk 2 above)
Comment (dy) [reply src/smart-app-banner.js:1]: Not convinced that replacing a file with a folder with the file enhances readability. We don’t have and not planning to spawn tenths of various modules, so until there is a tangible reason for that let’s keep things simple

File: package.json (hunk 3 above)
Comment (dy) [reply package.json:3]: `npm version` creates tag also.

File: src/smart-app-banner.scss (hunk 1 above)
Comment (dy) [reply src/smart-app-banner.scss:1]: I believe there is no good reason to include preprocessor, there is nothing which can’t be simply done with pure css.
Also see https://github.com/kudago/smart-app-banner/pull/42#issuecomment-191503326

Comment (ghost) [reply src/smart-app-banner.scss:1]: ok, I will remove it

**Pull request closed without merging**